pkg/modprovider/planstore.go

```diff
@@ -31,9 +31,41 @@ import (
 //
 // The provider will have a single planStore instance that stores the information indexed by URNs.
 type planStore struct {
-	mutex  sync.Mutex
-	plans  map[urn.URN]*planEntry
-	states map[urn.URN]*stateEntry
+	mutex     sync.Mutex
+	plans     map[urn.URN]*planEntry
+	states    map[urn.URN]*stateEntry
+	operation Operation
+}
+
+func (s *planStore) SetOperation(op Operation) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	s.operation = op
+}
+
+func (s *planStore) GetOperation() Operation {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	return s.operation
+}
+
+type Operation string
+
+const (
+	OperationDelete Operation = "DELETE"
+)
+
+// getPlanEntry returns the plan entry for the given URN.
+// If the plan entry does not exist, it returns nil and callers
+// are responsible for handling
+// NOTE: you probably want to use `getOrCreatePlanEntry` instead of this method.
```

Comment:
Consider returning (*planEntry, bool) with an OK flag to force caller to handle (nit).